qds_sdk/nezha.py

```diff
@@ -6,169 +6,251 @@
 
         update.set_defaults(func=NezhaCmdLine.update, default=None)
 
-
     @staticmethod
     def run(args):
-        a1 = args.pop(0)
-        NezhaCmdLine.classname = NezhaCmdLine.arg_to_classname[a1]
-        parser = NezhaCmdLine.parsers(a1)
+        parser = NezhaCmdLine.parsers()
         parsed = parser.parse_args(args)
+        NezhaCmdLine.classname = NezhaCmdLine.arg_to_classname[
+            parsed.nezha_entity]
         return parsed.func(globals()[NezhaCmdLine.classname], parsed)
 
     @staticmethod
     def update(cls, args):
         args_dict = {}
         for key, value in args.__dict__.items():
-            if key != "func" and key != "id" and value is not None:
+            params = (key != "func" and key != "id" and key != "nezha_entity")
+            if params and value is not None:
                 args_dict[key] = value
-        return json.dumps(globals()[NezhaCmdLine.classname].update(args.id, **args_dict), sort_keys=True, indent=4)
+
+        if args.nezha_entity == "default_datasource":
+            return json.dumps(
+                globals()[NezhaCmdLine.classname].update(**args_dict),
+                sort_keys=True, indent=4)
+
+        return json.dumps(
+            globals()[NezhaCmdLine.classname].update(args.id, **args_dict),
+            sort_keys=True, indent=4)
+
 
 class NezhaDataSource(Resource):
     """
-    qds_sdk.NezhaDataSource setting up Qubole Nezha class.
-    """
+    qds_sdk.NezhaDataSource: is the class for retrieving and
+    manipulating Qubole's NezhaDataSource.
 
-    """ endpoint for data source is  /nezha_data_sources endpoint"""
+    Endpoint for data source is: /nezha_data_sources.
+    """
 
     rest_entity_path = "nezha_data_sources"
 
+
 class NezhaPartition(Resource):
     """
-    qds_sdk.NezhaDataSource setting up Qubole Nezha class.
-    """
+    qds_sdk.NezhaPartition: is the class for retrieving and
+    manipulating Qubole's NezhaPartition.
 
-    """ endpoint for data source is  /nezha_data_sources endpoint"""
+    Endpoint for partitions is: /nezha_partitions.
+    """
 
     rest_entity_path = "nezha_partitions"
 
+
 class NezhaCube(Resource):
     """
-    qds_sdk.NezhaDataSource setting up Qubole Nezha class.
-    """
+    qds_sdk.NezhaCube: is the class for retrieving and
+    manipulating Qubole's NezhaCube.
 
-    """ endpoint for data source is  /nezha_data_sources endpoint"""
+    Endpoint for cubes is: /nezha_cubes.
+    """
 
     rest_entity_path = "nezha_cubes"
 
+
 class NezhaDimension(Resource):
     """
-    qds_sdk.NezhaDataSource setting up Qubole Nezha class.
-    """
+    qds_sdk.NezhaDimension: is the class for retrieving and
+    manipulating Qubole's NezhaDimension.
 
-    """ endpoint for data source is  /nezha_data_sources endpoint"""
+    Endpoint for dimensions is: /nezha_cubes/{nezha_cube_id}/nezha_dimensions.
+    User needs to set it before using.
+    """
 
     rest_entity_path = ""
```

Comment:
NezhaDimension/NezhaMeasure were needed in PyApi test and rest_entity_path needs NezhaCube Id. So, before using it there, we are setting rest_entity_path based on cube-id.
No command-line support is given for Dimensions/Measures.